### DISCUSSION TITLE
UtilitySlots & WingSlot Coordination

### DISCUSSION THREAD

**Discussion by wozniakty** [abluescarab/tModLoader-WingSlot#50] [Ideas]
Starting a thread for historical context on my mod and to share any ideas or feedback as time goes on.  Some background:

I absolutely love the WingSlot mod, particularly when playing with bigger mods like Thorium, Calamity, etc, there are so many cool items to try that feel build-defining, that using up a slot on a mandatory item like the Wings just feels like a downer.  My friend and I wanted to take that a step further and also give slots for boots and balloons (well, jump-related items, more on that later).

For anyone coming back to this in the future, the repo in question is my fork [here](https://github.com/wozniakty/tModLoader-UtilitySlots)

Per your feedback @abluescarab, I'll work on compatibility between our mods so that if both are installed everything works smoothly.  I think that's super reasonable expectation.  A question for you:
- Is there a way to _detect_ what other mods are installed and do it automatically?  I could certainly offer a config to turn off the wing slot in my mod manually, but automatic would certainly be better.

Some other random thoughts I've been noodling on with moving forward, if you have any ideas or feedback it would be appreciated but these aren't blockers or critical issues:
- How hard would it be to disable the modifiers from reforge on the items in these slots?  Would help with not overly impacting the balance _quite_ so much when you have 3 free accessories.
- I'm lucky that balloons are rendered so I can check the  `balloonSlot` property, but I think it might be even better to eventually include horseshoes and cloud in a bottle items in the list of valid items for that slot (thinking of renaming it to "Jump Slot" if I can get that working) but I can't find any clear unifying property between them.

**Comment by abluescarab** [comment]
Thanks for posting here! I detected whether another mod exists in MultiLure, but I'm not sure if that code still works in the new version of tModLoader. It would work something like this in your `ModName.cs`:
```
public override void PostSetupContent() {
    Mod wingSlotMod = ModLoader.GetMod("WingSlot");

    if(wingSlotMod != null) {
        // do stuff if mod exists
    }
}
```
I'm not sure how easy it would be to disable the modifiers, because the slot calls `VanillaUpdateAccessory()` and `VanillaUpdateEquip()`. You may need a GlobalItem override or something else. Additionally, if there is no clear unifying feature between each item, you will probably need to just check for each individual feature. If another open source mod adds any horseshoes or clouds in a bottle, you can check how those items are initialized to ensure you also cover any mod that might add one.

**Comment by wozniakty** [comment]
Ok perfect, thanks for the advice.  The code was all wired up and tested yesterday just waiting for an appropriate conditional to check against for the boolean, so this should work great.

I suppose I could always try clearing out the modifiers altogether on equip.  Sure it loses whatever roll was on them but that's probably acceptable enough if someone wants to play with that rule on.  It'd be a non-default config anyways.

Thanks again for the help, I'll share when it's updated and released.

**Comment by wozniakty** [comment]
Got it working, and officially published!  Having some issues with the automatic github actions publisher not finding the .NET assembly references but I suppose that's fine for now 🤷

**Reply by abluescarab** [reply]
Fantastic, glad to hear it!
